src/z3c/sqlalchemy/base.py

```diff
@@ -96,13 +96,13 @@ def __init__(self, dsn, model=None, transactional=True, twophase=False,
     @property
     def metadata(self):
         if not hasattr(self, '_v_metadata'):
-            self._v_metadata = MetaData(self._engine)
+            self._v_metadata = MetaData()
```

Comment:
See here: https://docs.sqlalchemy.org/en/20/errors.html#a-bind-was-located-via-legacy-bound-metadata-but-since-future-true-is-set-on-this-session-this-bind-is-ignored

Metadata is not using the engine anymore, the session has to.